localstack/services/kms/provider.py

```diff
@@ -133,89 +163,238 @@ def __init__(self, message=None):
-
-        return result
+        key = self._get_store(context).create_key(request, context.account_id, context.region)
+        return CreateKeyResponse(KeyMetadata=key.metadata)
+
+    @handler("ScheduleKeyDeletion", expand=False)
+    def schedule_key_deletion(
+        self, context: RequestContext, request: ScheduleKeyDeletionRequest
+    ) -> ScheduleKeyDeletionResponse:
+        pending_window = int(request.get("PendingWindowInDays", 30))
+        if pending_window < 7 or pending_window > 30:
+            raise ValidationException(
+                f"PendingWindowInDays should be between 7 and 30, but it is {pending_window}"
+            )
+        key = self._get_key(context, request.get("KeyId"))
+        key.schedule_key_deletion(pending_window)
+        attrs = ["DeletionDate", "KeyId", "KeyState"]
+        result = select_attributes(key.metadata, attrs)
+        result["PendingWindowInDays"] = pending_window
+        return ScheduleKeyDeletionResponse(**result)
+
+    @handler("CancelKeyDeletion", expand=False)
+    def cancel_key_deletion(
+        self, context: RequestContext, request: CancelKeyDeletionRequest
+    ) -> CancelKeyDeletionResponse:
+        key = self._get_key(context, request.get("KeyId"))
+        key.metadata["KeyState"] = "Disabled"
+        key.metadata["DeletionDate"] = None
+        # https://docs.aws.amazon.com/kms/latest/APIReference/API_CancelKeyDeletion.html#API_CancelKeyDeletion_ResponseElements
+        # "The Amazon Resource Name (key ARN) of the KMS key whose deletion is canceled."
+        return CancelKeyDeletionResponse(KeyId=key.metadata.get("Arn"))
+
+    @handler("DisableKey", expand=False)
+    def disable_key(self, context: RequestContext, request: DisableKeyRequest) -> None:
+        key = self._get_key(context, request.get("KeyId"))
+        key.metadata["KeyState"] = "Disabled"
+        key.metadata["Enabled"] = False
+
+    @handler("EnableKey", expand=False)
+    def enable_key(self, context: RequestContext, request: EnableKeyRequest) -> None:
+        key = self._get_key(context, request.get("KeyId"))
+        key.metadata["KeyState"] = "Enabled"
+        key.metadata["Enabled"] = True
+
+    @handler("ListKeys", expand=False)
+    def list_keys(self, context: RequestContext, request: ListKeysRequest) -> ListKeysResponse:
+        # https://docs.aws.amazon.com/kms/latest/APIReference/API_ListKeys.html#API_ListKeys_ResponseSyntax
+        # Out of whole KeyMetadata only two fields are present in the response.
+        keys_list = PaginatedList(
+            [
+                {"KeyId": key.metadata["KeyId"], "KeyArn": key.metadata["Arn"]}
+                for key in self._get_store(context).keys.values()
+            ]
+        )
+        page, next_token = keys_list.get_page(
+            lambda key_data: key_data.get("KeyId"),
+            next_token=request.get("Marker"),
+            page_size=request.get("Limit", 100),
```

Comment:
Is the default limit here from AWS, was it preexisting like that in the codebase or was it a self-chosen value?

Reply:
That is according to AWS docs: https://docs.aws.amazon.com/kms/latest/APIReference/API_ListKeys.html#API_ListKeys_RequestParameters
Added a comment about that.